.travis.yml

```diff
@@ -1,37 +1,86 @@
-dist: trusty
-
+dist: bionic
 language: cpp
 
-matrix:
+branches:
+  only:
+    - master
+    - development
+
+compiler:
+  - clang   # clang 7
+  - clang-9
+  - clang-10
+  - gcc     # gcc 7.4
+  - gcc-9
+  - gcc-10
+
+jobs:
   include:
-    - compiler: gcc
+    - compiler: clang
+    - name: "Build with LLVM/Clang 7"
+      env: 
+      - MATRIX_EVAL="CC=clang CXX=clang++"
+
+    - compiler: clang-9
+    - name: "Build with LLVM/Clang 9"
+      addons:
+        apt:
+          packages:
+          - clang-9
+      env: 
+      - MATRIX_EVAL="CC=clang-9 CXX=clang++-9"
+
+    - compiler: clang-10
+    - name: "Build with LLVM/Clang 10"
       addons:
         apt:
           sources:
-            - ubuntu-toolchain-r-test
+          - sourceline: 'deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main'
+          key_url: 'https://apt.llvm.org/llvm-snapshot.gpg.key'
           packages:
-            - g++-4.9
```

Comment:
I did remove the previous compiler build entries, but if they're still needed, I can re-add them.